src/commands/app/upgrade.tsx

```diff
@@ -221,3 +260,80 @@ async function forceTargetVersionSelection(
     targetAppVersionString,
   );
 }
+
+async function updateMissingSystemSoftwareDependency(
+  process: ProcessRenderer,
+  apiClient: MittwaldAPIV2Client,
+  appInstallationId: string,
+  dependency: AppSystemSoftwareDependency,
+  force: boolean,
+) {
+  const dependencySoftware = await apiClient.app.getSystemsoftware({
+    systemSoftwareId: dependency.systemSoftwareId,
+  });
+  assertSuccess(dependencySoftware);
+
+  const dependencyVersionList = await apiClient.app.listSystemsoftwareversions({
+    systemSoftwareId: dependency.systemSoftwareId,
+  });
+  assertSuccess(dependencyVersionList);
+
+  let dependencyTargetVersion: AppSystemSoftwareVersion = {
+    id: "not yet set",
+    externalVersion: "0.0.0",
+    internalVersion: "0.0.0",
+  };
+  for (const dependencyVersion of dependencyVersionList.data) {
+    if (
+      semver.satisfies(
+        dependencyVersion.internalVersion,
+        dependency.versionRange,
+      ) &&
+      semver.gt(
+        dependencyVersion.internalVersion,
+        dependencyTargetVersion.internalVersion,
+      )
+    ) {
+      dependencyTargetVersion = dependencyVersion;
+    }
+  }
+
+  if (!force) {
+    const confirmed: boolean = await process.addConfirmation(
+      <Text>
+        {dependencySoftware.data.name as string} will be updated to Version{" "}
+        {dependencyTargetVersion.externalVersion} - Continue?
+      </Text>,
+    );
+    if (!confirmed) {
+      process.addInfo(<Text>Upgrade will not be triggered.</Text>);
+      process.complete(<></>);
+      ux.exit(1);
+    }
+  } else {
+    process.addInfo(
+      <Text>
+        Commencing upgrade of {dependencySoftware.data.name as string} to{" "}
+        Version {dependencyTargetVersion.externalVersion}.
+      </Text>,
+    );
+  }
+
+  const systemSoftware: {
+    [x: string]: {
+      systemSoftwareVersion: string | undefined;
+    };
+  } = {};
+  systemSoftware[dependencySoftware.data.id as string] = {
+    systemSoftwareVersion: dependencyTargetVersion.id,
+  };
+
+  const data = { systemSoftware };
+
+  const dependencyUpgradeResponse = await apiClient.app.patchAppinstallation({
```

Comment:
Doesn't this result in a multitude of `patchAppinstallation` requests (once for the upgrade itself, and then once for each dependency)? Might it be more robust to aggregate all those in to a single `patchAppinstallation` request?

Reply:
Yeah it does, I changed the implementation to aggregate everything into one request.